pkg/detector/detector.go

```diff
@@ -226,10 +245,11 @@ func (d *ResourceDetector) Reconcile(key util.QueueKey) error {
 		return err
 	}
 	if propagationPolicy != nil {
-		// return err when dependents not present, that we can retry at next reconcile.
+		// Add clusterWideKey to waiting list when dependents not present, waiting for override to be created.
 		if present, err := helper.IsDependentOverridesPresent(d.Client, propagationPolicy); err != nil || !present {
-			klog.Infof("Waiting for dependent overrides present for policy(%s/%s)", propagationPolicy.Namespace, propagationPolicy.Name)
-			return fmt.Errorf("waiting for dependent overrides")
+			d.AddWaiting(clusterWideKey)
+			klog.Infof("object's key has been added to waiting list")
```

Comment:
I think we should keep the original log here, so that we are able to know the reason

And we should separate `err` and `present` here, for `err!=nil` we just return the error

Reply:
I think it should do the same check for `ClusterPropagationPolicy`, but it didn't, I guess it's an oversight, right? @RainbowMango 

If yes, we should watch `ClusterPropagationPolicy` as well